src/XMakeBuildEngine/Construction/ProjectElementContainer.cs

```diff
@@ -437,6 +476,36 @@ private void AddInitialChild(ProjectElement child)
 
             XmlElement.AppendChild(child.XmlElement);
 
+            if (XmlDocument.PreserveWhitespace)
+            {
+                //  If we are trying to preserve formatting of the file, then the new node won't automatically be indented.
+                //  So try to match the surrounding formatting and add one indentation level
+                if (XmlElement.FirstChild.NodeType == XmlNodeType.Whitespace)
+                {
+                    //  This container had a whitespace node, which should generally be a newline and the indent
```

Comment:
I thought about this and it seemed OK to leave it as is.  If there's spacing between the elements but not a newline, then we'll still be matching the surrounding formatting, even though the formatting seems weird.